docs/external-node.md

```diff
@@ -0,0 +1,565 @@
+
+   After preparing the `ExternalNode` configuration yaml for the VM, we can
+   apply it in the cluster.
+
+   ```bash
+   $ cat << EOF | kubectl apply -f -
+   apiVersion: crd.antrea.io/v1alpha1
+   kind: ExternalNode
+   metadata:
+     name: vm1
+     namespace: vm-ns
+     labels:
+       role: db
+   spec:
+     interfaces:
+       - ips: [ "172.16.100.3" ]
+         name: ""
+   EOF
+   ```
+
+### Installation on Linux VM
+
+#### Prerequisites on Linux VM
+
+OVS needs to be installed on the VM. For more information about OVS installation
+please refer to the [getting-started guide](getting-started.md#open-vswitch).
+
+#### Installation steps on Linux VM
+
+1. Build `antrea-agent` binary in the root of the antrea code tree and copy the
+   `antrea-agent` binary from the `bin` directory to the Linux VM.
+
+   ```bash
+   make docker-bin
+   ```
+
+2. The `antrea-agent.conf` file specifies agent configuration parameters. Copy
+   the [agent configuration file](../build/yamls/externalnode/conf/antrea-agent.conf)
+   to the VM and edit the `antrea-agent.conf` file to set `clientConnection`,
+   `antreaClientConnection` and `externalNodeNamespace` with the correct values.
+   Copy `antrea-agent.antrea.kubeconfig` and `antrea-agent.kubeconfig` files to
+   the VM, that were generated in the step 4 and step 5 of
+   [Prerequisites on Kubernetes cluster](#prerequisites-on-kubernetes-cluster).
+
+   ```bash
+   AGENT_NAMESPACE="vm-ns"
+   AGENT_CONF_PATH="/etc/antrea"
+   mkdir -p $AGENT_CONF_PATH
+   # Copy antrea-agent kubeconfig files
+   cp ./antrea-agent.kubeconfig $AGENT_CONF_PATH
+   cp ./antrea-agent.antrea.kubeconfig $AGENT_CONF_PATH
+   # Update clientConnection and antreaClientConnection
+   sed -i "s|kubeconfig: |kubeconfig: $AGENT_CONF_PATH/|g" antrea-agent.conf
+   sed -i "s|#externalNodeNamespace: default|externalNodeNamespace: $AGENT_NAMESPACE|g" antrea-agent.conf
+   # Copy antrea-agent configuration file
+   cp ./antrea-agent.conf $AGENT_CONF_PATH
+   ```
+
+3. Create `antrea-agent` service. Below is a sample snippet to start
+   `antrea-agent` as a service on Ubuntu 18.04 or later:
+
+   ```bash
+   AGENT_BIN_PATH="/usr/sbin"
+   AGENT_LOG_PATH="/var/log/antrea"
+   mkdir -p $AGENT_BIN_PATH
+   mkdir -p $AGENT_LOG_PATH
+   cat << EOF > /etc/systemd/system/antrea-agent.service
+   Description="antrea-agent as a systemd service"
+   After=network.target
+   [Service]
+   ExecStart=$AGENT_BIN_PATH/antrea-agent \
+   --config=$AGENT_CONF_PATH/antrea-agent.conf \
+   --logtostderr=false \
+   --log_file=$AGENT_LOG_PATH/antrea-agent.log
+   Restart=on-failure
+   [Install]
+   WantedBy=multi-user.target
+   EOF
+   
+   sudo systemctl daemon-reload
+   sudo systemctl enable antrea-agent
+   sudo systemctl start antrea-agent
+   ```
+
+4. Configure environment variable `NODE_NAME` if the VM's hostname is different
+   from the name defined in the `ExternalNode` resource.
+
+   ```bash
+   export NODE_NAME="vm1"
+   ```
```

Comment:
does this step come a bit late (why after 3.?). How do I set the env variable when I run antrea-agent as a systemd service?